scrapy/utils/misc.py

```diff
@@ -162,3 +164,49 @@ def set_environ(**kwargs):
                 del os.environ[k]
             else:
                 os.environ[k] = v
+
+
+_trailing_whitespaces_re = re.compile(r'^(\s*)')
+
+
+def scan_generator_for_return_statement(callable):
+    """
+    Returns True if a callable is a generator function and includes a
+    'return' statement with a value different than None, False otherwise
+    """
+
+    def reduce_indentation_level(callable_lines):
```

Comment:
I'm not really fond of this, suggestions are welcome.

Reply:
Exactly what I needed, thanks! Too bad it's not available on py2, but hopefully that won't be an issue for too long :crossed_fingers:

Reply:
Well, actually there's no need to worry about that, precisely because of https://github.com/scrapy/scrapy/pull/3869#issuecomment-510744562